### ISSUE TITLE
App stops when allowing permission to storage

### ISSUE THREAD

**Issue by CJM01** [jahirfiquitiva/Kuper#6]
###### Device/dashboard info:
 - Kuper version: `1.1.1`
 - Device OS version: `8.1`
 - Device Manufacturer: `HTC`
 - Device Name: `Marlin`

###### Reproduction Steps

1.  Download project, or sample app
2.  Make no edits and run through Android Studio, or install sample app
 

###### Expected Result
Successful build or install

###### Actual Result
Force close

###### Work around
If you deny permission to storage, the app will open.  But once permission is allowed, the app will force close

Video https://www.youtube.com/watch?v=1mcj-rhM-ng

**Comment by jahirfiquitiva** [comment]
Thanks for the report. Can you please add some logs of the crash?

**Comment by jacksonhvisuals** [comment]
Ran into this as well.
```bash
1-18 23:25:47.679 4784-4784/xyz.jacksonhayes.widgets.sunrise E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                Process: xyz.jacksonhayes.widgets.sunrise, PID: 4784
                                                                                java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=55, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {xyz.jacksonhayes.widgets.sunrise/xyz.jacksonhayes.widgets.sunrise.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.os.Bundle.unparcel()' on a null object reference
                                                                                    at android.app.ActivityThread.deliverResults(ActivityThread.java:4268)
                                                                                    at android.app.ActivityThread.handleSendResult(ActivityThread.java:4312)
                                                                                    at android.app.ActivityThread.-wrap19(Unknown Source:0)
                                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1644)
                                                                                    at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                    at android.os.Looper.loop(Looper.java:164)
                                                                                    at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                                    at java.lang.reflect.Method.invoke(Native Method)
                                                                                    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                                 Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.os.Bundle.unparcel()' on a null object reference
                                                                                    at android.os.Bundle.putAll(Bundle.java:288)
                                                                                    at android.content.Intent.putExtras(Intent.java:8712)
                                                                                    at jahirfiquitiva.libs.kuper.ui.activities.KuperActivity$setupContent$1$1.invoke(KuperActivity.kt:303)
                                                                                    at jahirfiquitiva.libs.kuper.ui.activities.KuperActivity$setupContent$1$1.invoke(KuperActivity.kt:60)
                                                                                    at jahirfiquitiva.libs.frames.ui.activities.base.BaseWallpaperActionsActivity.executeQueuedAction(BaseWallpaperActionsActivity.kt:100)
                                                                                    at jahirfiquitiva.libs.frames.ui.activities.base.BaseWallpaperActionsActivity.onRequestPermissionsResult(BaseWallpaperActionsActivity.kt:73)
                                                                                    at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:7419)
                                                                                    at android.app.Activity.dispatchActivityResult(Activity.java:7270)
                                                                                    at android.app.ActivityThread.deliverResults(ActivityThread.java:4264)
                                                                                    at android.app.ActivityThread.handleSendResult(ActivityThread.java:4312) 
                                                                                    at android.app.ActivityThread.-wrap19(Unknown Source:0) 
                                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1644) 
                                                                                    at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                    at android.os.Looper.loop(Looper.java:164) 
                                                                                    at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                                    at java.lang.reflect.Method.invoke(Native Method) 
                                                                                    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
01-18 23:25:47.795 4784-4795/xyz.jacksonhayes.widgets.sunrise I/zygote: NativeAlloc concurrent copying GC freed 7742(571KB) AllocSpace objects, 5(1200KB) LOS objects, 49% free, 3MB/6MB, paused 2.935ms total 117.767ms
01-18 23:25:47.848 4784-4789/xyz.jacksonhayes.widgets.sunrise I/zygote: Do full code cache collection, code=78KB, data=74KB
01-18 23:25:47.849 4784-4789/xyz.jacksonhayes.widgets.sunrise I/zygote: After code cache collection, code=74KB, data=43KB
01-18 23:25:47.897 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err: java.util.zip.ZipException: File too short to be a zip file: 0
01-18 23:25:47.897 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.zip.ZipFile.<init>(ZipFile.java:214)
01-18 23:25:47.898 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.zip.ZipFile.<init>(ZipFile.java:149)
01-18 23:25:47.898 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.zip.ZipFile.<init>(ZipFile.java:163)
01-18 23:25:47.898 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.kuper.providers.viewmodels.KuperViewModel.getWidgetPreviewsPathFromZip(KuperViewModel.kt:101)
01-18 23:25:47.899 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.kuper.providers.viewmodels.KuperViewModel.internalLoad(KuperViewModel.kt:52)
01-18 23:25:47.899 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.kuper.providers.viewmodels.KuperViewModel.internalLoad(KuperViewModel.kt:35)
01-18 23:25:47.899 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.archhelpers.viewmodels.BasicViewModel.safeInternalLoad(BasicViewModel.kt:67)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.archhelpers.viewmodels.BasicViewModel.access$safeInternalLoad(BasicViewModel.kt:25)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.archhelpers.viewmodels.BasicViewModel$loadData$1.doLoad(BasicViewModel.kt:42)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.archhelpers.tasks.Async$execute$1.invoke(Async.kt:35)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at jahirfiquitiva.libs.archhelpers.tasks.Async$execute$1.invoke(Async.kt:23)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at org.jetbrains.anko.AsyncKt$doAsync$1.invoke(Async.kt:140)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at org.jetbrains.anko.AsyncKt$doAsync$1.invoke(Unknown Source:0)
01-18 23:25:47.900 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at org.jetbrains.anko.AsyncKt$sam$Callable$761a5578.call(Unknown Source:2)
01-18 23:25:47.901 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
01-18 23:25:47.901 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
01-18 23:25:47.901 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
01-18 23:25:47.901 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
01-18 23:25:47.901 4784-4809/xyz.jacksonhayes.widgets.sunrise W/System.err:     at java.lang.Thread.run(Thread.java:764)
```

**Comment by jahirfiquitiva** [comment]
Thanks for the logs @jacksonhvisuals 
Will look into it asap.

![bitmoji](https://render.bitstrips.com/v2/cpanel/e4f91d37-026a-44d0-91ea-b26cfdaa34a7-f1a30f5f-4907-483f-874a-54b05c8475ab-v1.png?transparent=1&palette=1&width=246)

**Comment by jahirfiquitiva** [comment]
Should be fixed with version `1.1.2`
